Detectors/MUON/MCH/Raw/Encoder/Digit/digit2raw.cxx

```diff
@@ -0,0 +1,135 @@
+#include <fstream>
+#include <iostream>
+#include <limits>
+#include <string>
+
+namespace po = boost::program_options;
+using namespace o2::mch::raw;
+
+namespace
+{
+std::string asString(const o2::InteractionRecord& ir)
+{
+  return fmt::format("ORB {:6d} BC {:4d}",
+                     ir.orbit, ir.bc);
+}
+
+std::string asString(const o2::InteractionTimeRecord& ir)
+{
+  return asString(static_cast<o2::InteractionRecord>(ir));
+}
+} // namespace
+
+std::ostream& operator<<(std::ostream& os, const o2::mch::Digit& d)
+{
+  os << fmt::format("DE {:4d} PADUID {:8d} ADC {:6d} TS {:g}",
+                    d.getDetID(), d.getPadID(), d.getADC(),
+                    d.getTimeStamp());
+  return os;
+}
+
+void digit2raw(const std::string& input,
+               DigitEncoder& encoder,
+               PayloadPaginator paginate)
+{
+  o2::InteractionRecord ir;
+  std::vector<o2::mch::Digit> digits;
+  DigitReader dr(input.c_str());
+
+  while (dr.nextDigits(ir, digits)) {
+
+    std::vector<std::byte> buffer;
+
+    encoder(digits, buffer, ir.orbit, ir.bc);
+
+    paginate(buffer);
+  }
+}
+
+int main(int argc, char* argv[])
+{
+  po::options_description generic("options");
+  bool userLogic{false};
+  bool dummyElecMap{false};
+  std::string input;
+  po::variables_map vm;
+
+  // clang-format off
+  generic.add_options()
+      ("help,h", "produce help message")
+      ("userLogic,u",po::bool_switch(&userLogic),"user logic format")
+      ("dummyElecMap,d",po::bool_switch(&dummyElecMap),"use a dummy electronic mapping (for testing only)")
+      ("output-dir,o",po::value<std::string>()->default_value("./"),"output directory for file(s)")
+      ("input-file,i",po::value<std::string>(&input)->required(),"input file name")
+      ("configKeyValues", po::value<std::string>()->default_value(""), "comma-separated configKeyValues")
+      ("no-empty-hbf,e", po::value<bool>()->default_value(true), "do not create empty HBF pages (except for HBF starting TF)");
+  // clang-format on
+
+  po::options_description cmdline;
+  cmdline.add(generic);
+
+  po::store(po::command_line_parser(argc, argv).options(cmdline).run(), vm);
+
+  if (vm.count("help")) {
+    std::cout << generic << "\n";
+    return 2;
+  }
+
+  try {
+    po::notify(vm);
+  } catch (boost::program_options::error& e) {
+    std::cout << "Error: " << e.what() << "\n";
+    exit(1);
+  }
+
+  po::notify(vm);
+
+  o2::conf::ConfigurableParam::updateFromString(vm["configKeyValues"].as<std::string>());
+
+  if (dummyElecMap) {
+    std::cout << "WARNING: using dummy electronic mapping\n";
+  }
+
+  auto det2elec = (dummyElecMap ? createDet2ElecMapper<ElectronicMapperDummy>() : createDet2ElecMapper<ElectronicMapperGenerated>());
+
+  DigitEncoder encoder = createDigitEncoder(userLogic, det2elec);
+
+  auto solar2feelink = (dummyElecMap ? createSolar2FeeLinkMapper<ElectronicMapperDummy>() : createSolar2FeeLinkMapper<ElectronicMapperGenerated>());
+
+  o2::raw::RawFileWriter fw;
```

Comment:
Better create writer aware of detector it serves, then it will be automatically used for setting the DAQ SourceID when you will be ready for RDHv6.

Reply:
ok. will change it